genai_stack/vectordb/base.py

```diff
@@ -1,59 +1,33 @@
-from typing import List, Any
+from typing import Callable
+from pydantic import BaseModel
 
-from langchain.docstore.document import Document
-from langchain.embeddings import HuggingFaceEmbeddings
+from genai_stack.stack.stack_component import StackComponent, StackComponentConfig
+from genai_stack.vectordb.chromadb.constants import DEFAULT_SEARCH_OPTIONS
 
-from genai_stack.core import BaseComponent
-from genai_stack.utils.importing import import_class
-from genai_stack.constants.vectordb import VECTORDB_CONFIG_KEY
 
+class BaseVectorDBConfigModel(BaseModel):
+    """Base Data Model for VectorDB"""
 
-class BaseVectordb(BaseComponent):
-    module_name = "VectorDB"
-    config_key = VECTORDB_CONFIG_KEY
+    pass
 
-    def __init__(self, config: dict) -> None:
-        """
-        A wrapper around the weaviate-client and langchain's weaviate class
 
-        Args:
-            config: Pass the parsed config file into this class
-        """
-        super().__init__(name=self.module_name, config=config)
-        self.parse_config(self.config_key, self.required_fields)
+class BaseVectorDBConfig(StackComponentConfig):
+    """Base VectorDB Config Stack Component Config"""
 
-    def search(self, query: str) -> List[Document]:
-        raise NotImplementedError()
+    data_model = BaseVectorDBConfigModel
 
-    def create_client(self):
-        raise NotImplementedError()
 
-    def get_langchain_client(self):
-        raise NotImplementedError()
+class BaseVectorDB(StackComponent):
+    """Base VectorDB Stack Component"""
 
-    def get_langchain_memory_client(self):
+    config_class = BaseVectorDBConfig
+    _search_method: Callable = None
```

Comment:
Why this one is private method, can't user override this

Reply:
No it shouldn't be that's unused code : Removed it here [823e3cf](https://github.com/aiplanethub/genai-stack/pull/23/commits/823e3cf2106c0111d6e34b83d51806b6216b4e5e)